doc/ipythonconsole.rst

```diff
@@ -95,20 +149,22 @@ For information on the features, commands and capabilities built into IPython it
 Using UMR to reload changed modules
 ===================================
 
-When working with scripts and modules in an interactive session, Python only loads a module from its source file once, the first time it is ``import``-ed.
-During this first ``import``, the bytecode (``.pyc`` file) is generated if necessary and the imported module object is cached in ``sys.modules``.
-If you subsequently re-import the module anytime in the same session without Spyder, this cached code object will be used even if its source code (``.py{w}`` file) has changed in the meantime.
-While efficient for final production code, this behavior is often undesired when working interactively, such as when analyzing data or testing your own modules.
-In effect, you're left with no way to update or modify any already-imported modules, aside from manually removing the relevant ``.pyc`` files, or restarting the console entirely.
+When working in an interactive session, Python only loads a module from its source file once, the first time it is ``import``-ed.
 
-Fortunately, in Spyder, there's an easy solution: the :guilabel:`User Module Reloader` (UMR), a Spyder-exclusive feature that, when enabled, automatically reloads modules right in the existing IPython shell whenever they are modified and re-imported, without any of the downsides of the above workarounds.
-Even better, Spyder also loads the ``%autoreload`` magic by default into any kernels it starts, allowing changes in already imported modules to be automatically picked up the as soon as the modified file is saved, without any additional user action.
-With UMR enabled, you can test complex applications within the same IPython interpreter without having to restart it every time you make a change, saving large amounts of manual tedium and long restart times.
-Or, if you're analyzing data step by step using your own custom libraries, you can easily add or tweak a function in the latter and see the results reflected in the former, all without the overhead of reloading the data and re-running your whole script to restore your session to the same point.
+Spyder's :guilabel:`User Module Reloader` (UMR) automatically reloads modules right in your existing IPython consoles whenever they are modified and re-imported.
+With UMR enabled, you can test changes to your code without restarting the kernel.
 
-UMR is enabled by default, and will do its work automatically without user intervention, although it will provide you with a red ``Reloaded modules:`` message in the console listing the files it has refreshed when it activates.
+.. image:: images/console/console-reload-modules.png
```

Comment:
I know we talked about adding a screenshot for this, but it's really hard to understand what's happening on it. So I think we should remove it (sorry for making you waste your time @juanis2112).

Reply:
Instead, we could just show a tight crop of the "reloaded modules" indication (from this screenshot or elsewhere), so users know what to look for with that and it doesn't take up much space.

Reply:
2/3rds of this screenshot is completely blank, taking up vertical space in the docs, and the feature it intends to highlight only takes up a single line, so I'd recommend either a tight crop as I suggested above, e.g. something like this:

![image](https://user-images.githubusercontent.com/17051931/102963747-8702b800-44af-11eb-99f3-5fd6e037dc15.png)

Or at least collapsing the vertical size of the console to just show the region of interest, e.g.

![image](https://user-images.githubusercontent.com/17051931/102963854-cdf0ad80-44af-11eb-9778-3592b77be6b6.png)

Thanks.

Reply:
This was done, thanks.

Reply:
Looks good to me too. Thanks @juanis2112!